src/activities/share/share.js

```diff
@@ -0,0 +1,240 @@
+var items
+
+var savedTotalBoys
+var savedTotalGirls
+var savedTotalCandies
+var savedPlacedInGrils
+var savedPlacedInBoys
+var savedCurrentCandies
+
+function start(items_) {
+    items = items_
+    currentLevel = 0
+    initLevel()
+}
+
+function stop() {
+}
+
+function initLevel() {
+    items.bar.level = currentLevel + 1
+    var filename = "resource/board/"+ "board" + currentLevel + ".qml"
+    items.dataset.source = filename
+
+    setUp()
+}
+
+function setUp() {
+    var levelData = items.dataset.item
+
+    // use board levels
+    if (currentLevel < 7) {
+        items.totalBoys = levelData.levels[items.currentSubLevel].totalBoys
+        items.totalGirls = levelData.levels[items.currentSubLevel].totalGirls
+        items.totalCandies = levelData.levels[items.currentSubLevel].totalCandies
+
+        items.instruction.text = levelData.levels[items.currentSubLevel].instruction
+        items.nbSubLevel = levelData.levels.length
+
+
+        items.background.currentCandies = items.totalGirls * levelData.levels[items.currentSubLevel].placedInGirls +
+                items.totalBoys * levelData.levels[items.currentSubLevel].placedInBoys
+
+        items.background.placedInGirls = levelData.levels[items.currentSubLevel].placedInGirls
+        items.background.placedInBoys = levelData.levels[items.currentSubLevel].placedInBoys
+        items.background.showCount = levelData.levels[items.currentSubLevel].showCount
+
+
+        items.background.rest = items.totalCandies -
+                Math.floor(items.totalCandies / items.totalChildren) * (items.totalBoys+items.totalGirls)
+        items.basketWidget.element.opacity = (levelData.levels[items.currentSubLevel].forceShowBakset === true ||
+                                              items.background.rest !== 0) ? 1 : 0
+
+    // create random (guided) levels
+    } else {
+        // get a random number between 1 and max for boys, girls and candies
+        var maxBoys = levelData.levels[0].maxBoys
+        var maxGirls = levelData.levels[0].maxGirls
+        var maxCandies = levelData.levels[0].maxCandies
+
+        items.totalBoys = Math.floor(Math.random() * maxBoys) + 1
+        items.totalGirls = Math.floor(Math.random() * maxGirls) + 1
+        var sum = items.totalBoys + items.totalGirls
+        // use sum * 4 as top margin (max 4 candies per rectangle)
+        items.totalCandies = Math.floor(Math.random() * (4 * sum - sum + 1)) + sum
+
+        // stay within the max margin
+        if (items.totalCandies > maxCandies)
+            items.totalCandies = maxCandies
+
+        // grammer correction depending the number of children: 1 (singular) or more (plural)
+        var boys
+        var girls
+
+        if (items.totalBoys > 1)
+            boys = "boys"
+        else boys = "boy"
+
+        if (items.totalGirls > 1)
+            girls = "girls"
+        else girls = "girl"
+
+        items.instruction.text = qsTr("Place " + items.totalBoys + " " + boys + " and " +
```

Comment:
you should not handle translation like that here.
It should be like in https://github.com/gcompris/GCompris-qt/blob/master/src/activities/balancebox/editor/balanceboxeditor.js#L157